mysql-test/suite/clone/t/local_basic_gtid_compression.test

```diff
@@ -0,0 +1,122 @@
+
+# Assert that 6 transactions have been committed in total.
+--let $assert_text = COMMITTED_GTIDS SERVER_UUID:1-6
+--let $assert_cond = "[SELECT @@GLOBAL.GTID_EXECUTED]" = "$server_uuid:1-6"
+--source include/assert.inc
+
+# When binlog is enabled, GTIDs of InnoDB transactions are updated by gtid
+# persister thread and GTIDs of other transactions are updated on binlog
+# rotation. So, wait till the GTIDs of InnoDB transactions are updated by the
+# gtid persister thread.
+#
+# i.e, wait till the below state is reached.
+#
+# SELECT * FROM mysql.gtid_executed;
+# source_uuid	interval_start	interval_end
+# ------------------------------------------
+# server_uuid	             1	           1
+# server_uuid	             4	           4
+# server_uuid	             6	           6
+
+--let $wait_condition = SELECT count(*) = 3 FROM mysql.gtid_executed
+--source include/wait_condition.inc
+
+--let $assert_cond = COUNT(*) = 3 FROM mysql.gtid_executed
+--let $assert_text = The mysql gtid table is not compressed and has gaps.
+--source include/assert.inc
+
+--echo #
+--echo # 3. Install Clone Plugin, enable debug point to print compression info
+--echo #    and initiate Clone operation.
+
+# Install Clone Plugin
+--replace_result $CLONE_PLUGIN CLONE_PLUGIN
+--eval INSTALL PLUGIN clone SONAME '$CLONE_PLUGIN'
+
+# Enable debug point to print the compression info for validation purpose.
+--let $debug_point = print_gtid_compression_info
+--source include/add_debug_point.inc
+
+# Clone data
+--connection clone_conn_1
+--source ../include/clone_command.inc
+
+# During clone operation, persister thread, along with writing InnoDB GTIDs, it
+# also writes non-InnoDB GTIDs and even compresses it by creating an explicit
+# request.
+
+--echo #
+--echo # 4. Assert that table is compressed after clone operation.
+--let $wait_condition = SELECT count(*) = 1 FROM mysql.gtid_executed WHERE interval_start = 1 AND interval_end = 6
+--source include/wait_condition.inc
+
+--let $assert_cond = COUNT(*) = 1 FROM mysql.gtid_executed
+--let $assert_text = The mysql gtid table should have 1 row.
+--source include/assert.inc
+
+--echo #
+--echo # 5. Assert that the compression was done by the gtid persister thread.
+
+# Find the thread that recently compressed the mysql.gtid_executed table and assert that it is clone_gtid_thread.
+--let $assert_file=$MYSQLTEST_VARDIR/log/mysqld.1.err
+--let $assert_match= .*Compression done by persister thread, first gapless row = 1-6.*
+--let $assert_select= Compression done by persister thread, first gapless row = 1-6
+--let $assert_text= GTID table compression is done by persister thread.
+--source include/assert_grep.inc
+
+--echo #
+--echo # 6. Cleanup
+--let $debug_point = print_gtid_compression_info
+--source include/remove_debug_point.inc
+DROP TABLE t1;
+DROP TABLE t2;
+--force-rmdir $CLONE_DATADIR
+--source ../include/clone_connection_end.inc
+
+UNINSTALL PLUGIN clone;
```

Comment:
Use wait_until_count_sessions.inc and count_session.inc combo

Reply:
Done.